analyzere/base_resources.py

```diff
@@ -148,6 +149,13 @@ def to_dict(self):
             d[k] = to_dict(v)
         return d
 
+    def __hash__(self):
+        # Certainly not the most efficient hash, since anything without an ID
+        # has the same hash value.
+        if hasattr(self, "id"):
+            int(uuid.UUID(self.id))
+        return 0
```

Comment:
@pmdarrow Unfortunately, no. If you consider [how we implemented `__eq__`](https://github.com/analyzere/analyzere-python/blob/master/analyzere/base_resources.py#L128), using `id()` would not satisfy the requirement of a `__hash__` function:

>  Hashable objects which compare equal must have the same hash value

>